### PR TITLE
Hlb wrap more services

### PR DESCRIPTION
**Pull request by nickchapman-da**
- support checkpoints from: CommandCompletionService.completionStream
- support: CommandCompletionService.completionEnd
- add high level type type for LedgerOffset  (replacing odd special handling in AbstractLedgerTypes)
- modify type of services which are known to always return AbsOffset
- add tests, (aside: may have discovered bug in sandbox)

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/hs/bindings/src/DA/Ledger/AbstractLedgerTypes.hs [hunk 1]
```diff
@@ -12,6 +12,13 @@ import DA.Ledger.Types
 import Proto3.Suite.Types
 import qualified Data.Map as Map
 
+-- TODO: Make LedgerOffset a proper wrapped type in Types.s
```
Comment (neil-da) [inline language-support/hs/bindings/src/DA/Ledger/AbstractLedgerTypes.hs]: `Types.s` - typo?

File: language-support/hs/bindings/src/DA/Ledger/Services/CommandCompletionService.hs [hunk 2]
```diff
@@ -1,28 +1,26 @@
 -- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
-module DA.Ledger.Services.CommandCompletionService (completionStream) where
+{-# LANGUAGE GADTs #-}
```
Comment (neil-da) [inline language-support/hs/bindings/src/DA/Ledger/Services/CommandCompletionService.hs]: Is this required?

File: language-support/hs/bindings/src/DA/Ledger/Types.hs [hunk 3]
```diff
@@ -50,6 +51,7 @@ module DA.Ledger.Types( -- High Level types for communication over Ledger API
 import Data.Map (Map)
 import Data.Text.Lazy (Text)
 import qualified Data.Text.Lazy as Text(unpack)
+--import DA.Ledger.AbstractLedgerTypes
```
Comment (neil-da) [inline language-support/hs/bindings/src/DA/Ledger/Types.hs]: No need to comment out imports

File: language-support/hs/bindings/src/DA/Ledger/AbstractLedgerTypes.hs (hunk 1 above)
Comment (nickchapman-da) [reply language-support/hs/bindings/src/DA/Ledger/AbstractLedgerTypes.hs]: yes

File: language-support/hs/bindings/src/DA/Ledger/Services/CommandCompletionService.hs (hunk 2 above)
Comment (nickchapman-da) [reply language-support/hs/bindings/src/DA/Ledger/Services/CommandCompletionService.hs]: Yes. The `ClientResult` type form the grpc library is a GADT. This is patterns matched in this file at line 60.

File: language-support/hs/bindings/src/DA/Ledger/Types.hs (hunk 3 above)
Comment (nickchapman-da) [reply language-support/hs/bindings/src/DA/Ledger/Types.hs]: yup

**Pull request merged**